source/blog/2019-03-02-update-on-module-unification-and-octane.md

```diff
@@ -0,0 +1,496 @@
+
+Discovering problems in a design you've been working on for so long is
+painful, especially in cases like this where the majority of the work was
+already complete, and we thought we were so close to the finish line.
+
+Programming language and API design is really hard. Sometimes I read old RFCs
+and marvel at how obvious the solution seems now, in contrast to the weeks,
+months, or years I know it took to tease it out from the millions of possible
+alternatives.
+
+When you're trying to balance so many competing constraints, sometimes a
+small change is all it takes to get you out of a design conundrum you've
+struggled with for months. In this case, that change was angle bracket
+component invocation.
+
+One thing making JavaScript imports difficult to use with templates was the
+constraint that components had to have a dash in their name. Unfortunately,
+dashes aren't valid in JavaScript identifiers, so something like `import
+some-component from './some-component'` produces a syntax error.
+
+Angle bracket components, on the other hand, start with a capital letter to
+disambiguate themselves from normal HTML tags: `<MyComponent />` instead of
+`{{my-component}}`. Most importantly, there's no dash.
+
+As the Ember community started using angle bracket syntax, early feedback was
+very positive. All of a sudden, JavaScript import syntax was back on the
+table.
+
+### The Path Forward: Template Imports
+
+Speaking for the Ember.js core team, we are trying to get better at updating
+the community when plans have changed but the new plan isn't fully locked in
+yet. So, consider this one of those situations.
+
+We know that the exact plan for Module Unification (MU), as described in the
+original RFC, will need to change. How it changes is not yet certain,
+but we believe that some of the problems we wanted to solve with MU are
+better solved with template imports.
+
+With template imports, we intend to make templates play nicely with
+JavaScript, so you can use the `import` feature you already know and love. By
+having components and helpers be modules you can import, we can eliminate the
+most complex parts of Module Unification so it's easier to learn and use.
+
+We recently posted the [SFC & Template Imports RFC][template-imports-rfc],
+which describes some of the low-level APIs needed in Ember to make template
+imports possible.
+
+[template-imports-rfc]: https://github.com/emberjs/rfcs/pull/454
+
+Learning from past mistakes, this RFC focuses on the primitives so we can
+quickly experiment, get feedback, and iterate on template import proposals in
+addons, before stabilizing them in the core framework.
+
+While the Ember.js core team has reached consensus that template imports are
+the intended path forward, please note that the current RFC only covers
+low-level primitives, not the API that would be used by Ember developers to
+author templates.
+
+Here is one example of a **very hypothetical** template imports syntax:
```

Comment:
Personally, I would _still_ prefer to see the "purest" form of template imports (strawman syntax is the `---` one) that does not mix with JS, before showing the full SFC example:

> Here is one example of a **very hypothetical** template imports syntax:
> 
> ```hbs
> ---
> import UserProfile from './user-profile';
> import UserIcon from './icons/user';
> ---
> <h1>{{this.blog.title}}</h1>
> <UserIcon />
> <UserProfile @userId={{this.blog.authorId}} />
> ```
> 
> The syntax is up in the air and will almost certainly be different from
> this example. Regardless, it shows the benefit of template imports clearly:
> we've imported two components—`UserProfile` and `UserIcon`— just like
> how we would refer to any other JavaScript module. This makes it very easy
> for everyone— from developer who are new to Ember, to IDEs, and other
> tooling in the JavaScript ecosystem`— to understand what came from where.
> 
> You can even imagine a (again, **very hypothetical**) single-file component
> format that places the template right within the component's class. Here, a
> unified imports solution feels especially natural:
> 
> ```javascript
> // src/ui/components/blog-post.gbs
> 
> import Component from '@glimmer/component';
> import UserIcon from './icons/user';
> 
> export class BlogPost extends Component {
>   blog = { title: 'Coming soon in Octane', authorId: '1234' };
> 
>   <template>
>     <h1>{{this.blog.title}}</h1>
>     <UserIcon />
>     <UserProfile @userId={{this.blog.authorId}} />
>   </template>
> }
> ```
> 
> Again, the exact syntax is up in the air and will almost certainly be different
> from this example. The benefit of exposing the low-level primitives first is that
> we can try out multiple competing designs relatively easily before comitting.
> And if you don't like what we eventually decide on, you can build an
> alternative that is just as stable as the default implementation.
> 
> ...

My personal preference aside, I genuienly think this separation/staging makes it much more easier to understand what is being talked about here, especially for most Ember developers who have not seen any of these yet. I mostly just copy and pasted your wording, which made it a bit repetitive, so feel free to edit it down.